scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamBackend.scala

```diff
@@ -116,7 +134,7 @@ object BeamPlanner {
             val ops: Seq[BeamOp[(K, Any)]] = cg.inputs.map(tp => rec(tp))
             CoGroupedOp(cg, ops)
           }
-          go(cg)
+          if (cg.descriptions.isEmpty) go(cg) else go(cg).withName(cg.descriptions.last)
```

Comment:
why using `last` here but `head` above?

Reply:
I meant to ask you about this myself. If you look at the test case in this PR, The cogrouped expression has two descriptions "Count words" and "Join with t1", both of which appear in the descriptions of `CoGrouped`. But since this appears to be appending descriptions for eg. in `CoGrouped.Pair` and the optimization rule `ComposeDescriptions`, I chose to keep the last one. Hence no assertion for "Count words" even in that test case.

Reply:
Why not combine all of them? Why not just `cg.descriptions.mkString("\n")` or something?

Reply:
Yes, let me try this out on a job first. Thanks!